testsuite/tests/typing-jkind-bounds/composite.ml

```diff
@@ -690,14 +693,22 @@ type 'a t = None | Some of ('a * 'a) t
 |}]
 
 let foo (t : _ t @@ contended) = use_uncontended t
-(* CR layouts v2.8: fix this *)
 [%%expect {|
 Line 1, characters 49-50:
 1 | let foo (t : _ t @@ contended) = use_uncontended t
                                                      ^
 Error: This value is "contended" but expected to be "uncontended".
 |}]
 
+let foo (t : int t @@ contended) = use_uncontended t
+(* CR layouts v2.8: this should work when we get tuples working *)
+[%%expect {|
+Line 1, characters 51-52:
+1 | let foo (t : int t @@ contended) = use_uncontended t
+                                                       ^
+Error: This value is "contended" but expected to be "uncontended".
+|}]
+
```

Comment:
I actually think that both these cases should be accepted, independently of tuples being accepted. Since a `'a t` can never contain a `'a`, its kind should always be `immutable_data`.